apps/greencheck/domain_check.py

```diff
@@ -127,16 +129,42 @@ def asn_from_ip(self, ip_address):
 
     def convert_domain_to_ip(
         self, domain
-    ) -> ipaddress.IPv4Network or ipaddress.IPv6Network or None:
+    ) -> typing.Union[ipaddress.IPv4Address, ipaddress.IPv6Address]:
         """
         Accepts a domain name or IP address, and returns an IPV4 or IPV6
-        address
+        address, raising an exception if not resolution occurs.
         """
-        try:
-            ip_string = socket.gethostbyname(domain)
-            return ipaddress.ip_address(ip_string)
-        except socket.gaierror as err:
-            logger.warning(f"Unable to lookup domain: {domain} - error: {err}")
+
+        # TODO: support multiple addresses being returned:
+        # `getaddrinfo`` actually returns a list of possible addresses,
+        # but our current code only assumes a domain would resolve to a single IP
+        # address when we look up a domain.
+        # Ideally we'd check that ALL the IP addresses resolved are within our
+        # green IP ranges but until we know how much this impacts performance
+        # we choose the first one.
+        ip_info = socket.getaddrinfo(domain, None)
```

Comment:
As the comments around this say - we acknowledge that we really should check that every ip range is green. Ths priority is fixing this first, before adding that functionality in later.